### PR TITLE
Keymap for XD75 with 7U spacebar EN-RU gamers

### PR DESCRIPTION
**Pull request by buzzlighter1**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xd75/keymaps/buzzlighter1/keymap.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+void led_set_user(uint8_t usb_led) {
+    if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
+        capslock_led_on();
+    } else {
+        capslock_led_off();
+    }
+}
+
+// Runs whenever there is a layer state change.
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (drashna) [inline keyboards/xd75/keymaps/buzzlighter1/keymap.c]: This should be changed:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

This allows the layer bitmask to be changed in size, if needed/wanted

File: keyboards/xd75/keymaps/buzzlighter1/keymap.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  uint8_t layer = biton32(state);
```
Comment (drashna) [inline keyboards/xd75/keymaps/buzzlighter1/keymap.c]:
```suggestion
  uint8_t layer = get_highest_layer(state);
```

File: keyboards/xd75/keymaps/buzzlighter1/layers.json [hunk 3]
```diff
@@ -0,0 +1 @@
..."KC_O", "KC_P", "KC_LBRC", "KC_RBRC", "KC_BSLS", "KC_PAUS", "KC_CAPS", "KC_A", "KC_S", "KC_D", "KC_F", "KC_G", "KC_H", "KC_J", "KC_K", "KC_L", "KC_SCLN", "KC_QUOT", "KC_NO", "KC_ENT", "KC_NO", "KC_LSFT", "KC_Z", "KC_X", "KC_C", "KC_V", "KC_B", "KC_N", "KC_M", "KC_COMM", "KC_DOT", "KC_SLSH", "KC_RSFT", "MO(1)", "KC_UP", "KC_NO", "KC_LCTL", "MO(0)", "KC_LALT", "KC_NO", "KC_SPC", "KC_NO", "KC_HOME", "KC_END", "KC_PGUP", "KC_PGDN", "TG(1)", "KC_RCTL", "KC_LEFT", "KC_DOWN", "KC_RGHT"], ["KC_NO", "KC_F1", "KC_F2", "KC_F3", "KC_F4", "KC_F5", "KC_F6", "KC_F7", "KC_F8", "KC_F9", "KC_F10", "KC_F11", "KC_F12", "KC_NO", "KC_NO", "KC_NO", "RGB_TOG", "RGB_MOD", "RGB_RMOD", "RGB_VAI", "RGB_VAD", "KC_P7", "KC_P8", "KC_P9", "KC_PMNS", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "BL_TOGG", "BL_INC", "BL_DEC", "BL_STEP", "KC_NO", "KC_P4", "KC_P5", "KC_P6", "KC_PPLS", "KC_NLCK", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_P1", "KC_P2", "KC_P3", "KC_PSLS", "KC_PENT", "KC_NO", "KC_TRNS", "KC_NO", "KC_NO", "KC_NO", "KC_TRNS", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_P0", "KC_PDOT", "KC_PAST", "KC_NO", "KC_NO", "KC_NO", "KC_NO", "KC_NO"]]
```
Comment (noroadsleft) [inline keyboards/xd75/keymaps/buzzlighter1/layers.json]: This file should be deleted IMO. It serves no purpose in this state.

If you use the command line, you can run `git rm --cached -- keyboards/xd75/keymaps/buzzlighter1/layers.json` to remove this file from version control, but keep it on your system.

**Comment by noroadsleft** [comment]
To elaborate on my comment regarding the `layers.json` file:

That file contains the contents of the key assignments that are sent to the QMK Compiler API when a compile job is launched in QMK Configurator. When that happens, the keyboard, keymap name, and layout macro used are provided separately from this data. The file contains no information on its own that can be imported back into QMK Configurator. The only way to get further use from this file is to edit it by hand in a text editor, providing the keyboard, keymap name, and layout macro that were used on the original compile job.

If you were to export the keymap JSON from Configurator at the time of compiling, and compare it to the `layers.json` file, you will see that `layers.json` is a subset of the data in the export, while the export is the complete data set, and can be imported back into Configurator.

**Comment by buzzlighter1** [comment]
I suspected something like that about the layers.json file, thank you the explanation!

My goal was to make the keymap available to the public. I have followed the guide in the link mentioned above, but I thought I need to merge it into qmk:master to open it for the public. If the keymap in the buzzlighter1:master serves the goal then I probably should cancel this merge?...

**Comment by noroadsleft** [comment]
Merging your keymap to qmk:master does make it more visible in my opinion, and later users may find something in your keymap they like.

If you want to include the JSON export from Configurator, that's fine and allowed, but it should be the `keymap.json` file with the complete data set, rather than the `layers.json` file.

Also, QMK's make system has the ability to compile `keymap.json` files into `keymap.c` files, so for ease of use, I'd suggest naming the `keymap.json` file to something more like `buzzlighter1_keymap.json`, because our CI system will compile the JSON instead of the C file if it finds both `keymap.c` and `keymap.json` in the same folder, and the C file supports more features.

There are some esoteric issues with Configurator `keymap.json` files, as far as compiling them through QMK Configurator. [Those issues are documented here.](https://github.com/qmk/qmk_configurator/issues/618#issuecomment-564955293) (I really should make a doc for this.) Also, the first part of that post is also out of scope for your specific use case; you should scroll down to "You can't use `default` as the value for the `keymap` key" and read from there.

**Comment by noroadsleft** [comment]
@buzzlighter1 If you want to contribute a QMK Configurator keymap file, additional pull requests are accepted.

I'm going to merge this; you should [resynchronize your `master` branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) now.

Thanks!

**Pull request merged**